### PR TITLE
Generate ReleasePlanAdmission for components

### PR DESCRIPTION
**Pull request by creydr**
Generate the RPA for the components
Relevant commit: ee71285f4798ade5d87498ca97ef674fc0ae3306 (+ c99b66e77007d93515fd53d9267793ce8bf280e4 and 1bacf859bed49ff80f832e5ab7337c61e91f84e7 to exclude releaseplanadmissions from konflux-apply)

The generated RPAs look for example like
```
apiVersion: appstudio.redhat.com/v1alpha1
kind: ReleasePlanAdmission
metadata:
  name: serverless-operator-135
  namespace: rhtap-releng-tenant
spec:
  applications: [serverless-operator-135]
  origin: ocp-serverless-tenant
  policy: registry-standard
  data:
    mapping:
      components:
        - name: serverless-openshift-kn-operator-135
          repository: registry.redhat.io/openshift-serverless-1/serverless-openshift-kn-operator
          pushSourceContainer: false
        - name: serverless-kn-operator-135
          repository: registry.redhat.io/openshift-serverless-1/serverless-kn-operator
          pushSourceContainer: false
        - name: serverless-ingress-135
          repository: registry.redhat.io/openshift-serverless-1/serverless-ingress
          pushSourceContainer: false
        - name: kn-serving-queue-115
          repository: registry.redhat.io/openshift-serverless-1/kn-serving-queue
          pushSourceContainer: false
        - name: kn-serving-activator-115
          repository: registry.redhat.io/openshift-serverless-1/kn-serving-activator
          pushSourceContainer: false
        - name: kn-serving-autoscaler-115
          repository: registry.redhat.io/openshift-serverless-1/kn-serving-autoscaler
          pushSourceContainer: false
        - name: kn-serving-autoscaler-hpa-115
          repository: registry.redhat.io/openshift-serverless-1/kn-serving-autoscaler-hpa
          pushSourceContainer: false
        ...
    defaults:
      tags:
        - "{{ git_sha }}"
        - "{{ git_short_sha }}"
        - "v1.35.0"
```

/hold
as we probably need to exclude it from the konflux-apply

### CODE REVIEW

**Comment by creydr** [comment]
/unhold
as we exclude the releaseplanadmissions from konflux-apply

File: cmd/generate/generate.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"embed"
 	"errors"
 	"fmt"
+	"github.com/openshift-knative/hack/pkg/util"
```
Comment (pierDipi) [inline cmd/generate/generate.go:7]: sort import?

File: cmd/konflux-apply/main.go [hunk 2]
```diff
@@ -2,7 +2,8 @@ package main
 
 import (
 	"context"
-	"flag"
+	"github.com/openshift-knative/hack/pkg/util"
+	"github.com/spf13/pflag"
 	"log"
```
Comment (pierDipi) [inline cmd/konflux-apply/main.go:7]: sort import?

File: pkg/konfluxgen/konfluxgen.go [hunk 3]
```diff
@@ -6,6 +6,9 @@ import (
 	"embed"
 	"encoding/json"
 	"fmt"
+	"github.com/coreos/go-semver/semver"
+	"github.com/openshift-knative/hack/pkg/soversion"
+	"github.com/openshift-knative/hack/pkg/util"
 	"io/fs"
```
Comment (pierDipi) [inline pkg/konfluxgen/konfluxgen.go:12]: sort imports?

File: cmd/generate/generate.go (hunk 1 above)
Comment (creydr) [reply cmd/generate/generate.go]: :+1: 
2a4669a592127e5c393e34db1119198958199a6e

File: cmd/konflux-apply/main.go (hunk 2 above)
Comment (creydr) [reply cmd/konflux-apply/main.go]: :+1: 
2a4669a592127e5c393e34db1119198958199a6e

File: pkg/konfluxgen/konfluxgen.go (hunk 3 above)
Comment (creydr) [reply pkg/konfluxgen/konfluxgen.go]: :+1: 
2a4669a592127e5c393e34db1119198958199a6e

File: pkg/konfluxgen/releaseplanadmission.template.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  applications: [{{{ truncate ( sanitize .ApplicationName ) }}}]
+  origin: ocp-serverless-tenant
+  policy: registry-standard
+  data:
+    mapping:
+      components:
+      {{{- range $component := .Components }}}
+        - name: {{{ $component.ComponentName }}}
+          repository: {{{ $component.ImageRepository }}}
+          pushSourceContainer: false
```
Comment (pierDipi) [inline pkg/konfluxgen/releaseplanadmission.template.yaml:16]: why false? Pushing source images can help for disaster recovery

File: pkg/prowgen/prowgen.go [hunk 5]
```diff
@@ -156,18 +156,18 @@ func Main() {
...
+		}
+		if err := runJobConfigInjectors(inConfigs, openShiftRelease); err != nil {
+			log.Fatalln("Failed to inject Slack reporter", err)
+		}
+		if err := RunOpenShiftReleaseGenerator(ctx, openShiftRelease); err != nil {
+			log.Fatalln("Failed to run openshift/release generator after injecting Slack reporter", err)
+		}
+		if err := PushBranch(ctx, openShiftRelease, remote, *branch, "Sync Serverless CI "+*inputConfig); err != nil {
+			log.Fatalln("Failed to push branch to openshift/release fork", *remote, err)
+		}*/
```
Comment (pierDipi) [inline pkg/prowgen/prowgen.go:172]: uncomment?

File: pkg/prowgen/prowgen_konflux.go [hunk 6]
```diff
@@ -326,6 +327,11 @@ func GenerateKonfluxServerlessOperator(ctx context.Context, openshiftRelease Rep
 			return fmt.Errorf("failed to generate Konflux configurations for %s (%s): %w", r.RepositoryDirectory(), branch, err)
 		}
 
+		csvPath := filepath.Join(r.RepositoryDirectory(), "olm-catalog", "serverless-operator", "manifests", "serverless-operator.clusterserviceversion.yaml")
+		if err := konfluxgen.GenerateReleasePlanAdmission(csvPath, resourceOutputPath, appName, semver.New(soMetadata.Project.Version)); err != nil {
+			return fmt.Errorf("failed to generate ReleasePlanAdmission: %w", err)
+		}
```
Comment (pierDipi) [inline pkg/prowgen/prowgen_konflux.go:333]: Can we encapsulate this into the `Generate` function ?

We could accept a CSV path file array config and generate those only when it's passed

File: pkg/prowgen/prowgen.go (hunk 5 above)
Comment (creydr) [reply pkg/prowgen/prowgen.go:172]: :facepalm:

File: pkg/konfluxgen/releaseplanadmission.template.yaml (hunk 4 above)
Comment (creydr) [reply pkg/konfluxgen/releaseplanadmission.template.yaml:16]: I wasn't sure if we need them in the production registry. Let me change it then...

File: pkg/prowgen/prowgen_konflux.go (hunk 6 above)
Comment (creydr) [reply pkg/prowgen/prowgen_konflux.go]: Not sure, why you mean a file array?
0d680d18

File: pkg/konfluxgen/releaseplanadmission.template.yaml (hunk 4 above)
Comment (creydr) [reply pkg/konfluxgen/releaseplanadmission.template.yaml]: 31acfba6

File: pkg/prowgen/prowgen.go (hunk 5 above)
Comment (creydr) [reply pkg/prowgen/prowgen.go]: 5aeccc15

**Comment by creydr** [comment]
rebased

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**